src/txn/txn_rollback_to_stable.c

```diff
@@ -110,20 +45,200 @@ __txn_abort_newer_update(
             upd->durable_ts = upd->start_ts = WT_TS_NONE;
         }
     }
+
+    /* Reset the history store flag for the stable update. */
+    if (first_upd != NULL)
+        F_CLR(first_upd, WT_UPDATE_HS);
+
+    return (first_upd);
 }
 
 /*
  * __txn_abort_newer_insert --
- *     Apply the update abort check to each entry in an insert skip list
+ *     Apply the update abort check to each entry in an insert skip list and return the first
+ *     satisfy update.
  */
-static void
+static WT_UPDATE *
 __txn_abort_newer_insert(
   WT_SESSION_IMPL *session, WT_INSERT_HEAD *head, wt_timestamp_t rollback_timestamp)
 {
     WT_INSERT *ins;
 
     WT_SKIP_FOREACH (ins, head)
-        __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
+        if (ins->upd != NULL &&
+          __txn_abort_newer_update(session, ins->upd, rollback_timestamp) != NULL)
+            return (ins->upd);
+
+    return (NULL);
+}
+
+/*
+ * __txn_abort_row_replace_with_hs_value --
+ *     Abort updates in the history store and replace the on-disk with an update that satisfy than
+ *     the given timestamp.
```

Comment:
Again, I am not sure what the comment means here. Please reword.

Reply:
The change is removed, so the comment also reverted to the original.